lib/Star/Kata/Infrastructure/InMemoryInfrastructure.php

```diff
@@ -0,0 +1,54 @@
+<?php
+/**
+ * This file is part of the phpkata project.
+ *
+ * (c) Yannick Voyer (http://github.com/yvoyer)
+ */
+
+namespace Star\Kata\Infrastructure;
+
+use Star\Kata\KataDomain\KataService;
+use Star\Kata\Model\KataCollection;
+use Star\Kata\Model\KataRepository;
+
+/**
+ * Class InMemoryInfrastructure
+ *
+ * @author  Yannick Voyer (http://github.com/yvoyer)
+ *
+ * @package Star\Kata\Infrastructure
+ */
+final class InMemoryInfrastructure implements KataInfrastructure
```

Comment:
I'm trying to define a boundary between an application and the domain logic. The infrastructure interface is only defining the needed objects the domain services need to work.

`InMemoryInfrastructure` defines all the repos used as in memory, compared to another implementation could be `DoctrineOrmInfrastructure`. `CacheInfrastructure`, FileSystemInfrastructure`  etc.